### PR TITLE
Don't depend on redhat-appstudio/application-api

### PR DESCRIPTION
**Pull request by zregvart**
The `github.com/redhat-appstudio/application-api` has older dependencies that are not compatible with the dependencies that we need. This tries to solve that issue by vendoring the specific types we need in the new `application` module.

Ref. https://issues.redhat.com/browse/HACBS-2426

### CODE REVIEW

File: .github/workflows/codeql.yaml [hunk 1]
```diff
@@ -70,8 +70,9 @@ jobs:
       - name: Download go dependencies
         run: |
           go mod download
-          go mod download -modfile tools/go.mod
-          go mod download -modfile acceptance/go.mod
+          (cd tools && go mod download)
```
Comment (lcarva) [inline .github/workflows/codeql.yaml:73]: Just for my own understanding, why not use `-modfile`?

File: application/v1alpha1/README [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The files in this directory were copied from
+https://github.com/redhat-appstudio/application-api/tree/main/api/v1alpha1
```
Comment (lcarva) [inline application/v1alpha1/README:2]: As a follow up, we could probably create a little script that automates this process. Then, we can run as a GitHub Workflow, warning us if any changes are found.

File: hack/boilerplate.go.txt
Comment (lcarva) [inline hack/boilerplate.go.txt:1]: What's this for?

**Review by lcarva — APPROVED** [review]

**Review by simonbaird — APPROVED** [review]
Idk, but lgtm.

**Comment by lcarva** [comment]
I think we may want to isolate the usage of the structs from Application API and instead provide our own structs that can be used more deliberately. A user coming to EC has no idea what an ApplicationSnapshotSpec is. I don't really want to send them to Appstudio docs for that. We should own that data structure here and have adapters/converters for the structs from Application API. Thoughts?

File: .github/workflows/codeql.yaml (hunk 1 above)
Comment (zregvart) [reply .github/workflows/codeql.yaml:73]: [This](https://github.com/enterprise-contract/ec-cli/actions/runs/5761947101/job/15620788857?pr=902#step:7:7) happens:
```
go: github.com/enterprise-contract/ec-cli/application@v0.0.0-00010101000000-000000000000 (replaced by ../application/): reading ../application/go.mod: open /home/runner/work/ec-cli/application/go.mod: no such file or directory
```

File: hack/boilerplate.go.txt
Comment (zregvart) [reply hack/boilerplate.go.txt:1]: Leftover from code generators, will remove

File: application/v1alpha1/README (hunk 2 above)
Comment (zregvart) [reply application/v1alpha1/README:2]: Created [HACBS-2441](https://issues.redhat.com/browse/HACBS-2441), I think we want to have an approach that is triggered on version updates and doesn't block us in CI

**Comment by zregvart** [comment]
> I think we may want to isolate the usage of the structs from Application API and instead provide our own structs that can be used more deliberately. A user coming to EC has no idea what an ApplicationSnapshotSpec is. I don't really want to send them to Appstudio docs for that. We should own that data structure here and have adapters/converters for the structs from Application API. Thoughts?

How would we maintain compatibility with App Studio in this scenario?

**Review by lcarva — APPROVED** [review]

**Pull request merged**